src/extremeweatherbench/evaluate.py

```diff
@@ -260,23 +260,39 @@ def compute_case_operator(
 ) -> pd.DataFrame:
     """Compute the results of a case operator.
 
-    This method will compute the results of a case operator. It will build
-    the target and forecast datasets,
-    align them, compute the metrics, and return a concatenated dataframe of the results.
+    This method will compute the results of a case operator. It validates
+    that all metrics are properly instantiated, builds the target and forecast
+    datasets, aligns them, and computes each metric with appropriate variable
+    pairs. Metrics with their own forecast_variable and target_variable use
+    only those variables; metrics without use all InputBase variable pairs.
 
     Args:
         case_operator: The case operator to compute the results of.
-        cache_dir: The directory to cache the mid-flight outputs of the workflow if
-        in serial mode.
+        cache_dir: The directory to cache mid-flight outputs (serial mode).
         kwargs: Keyword arguments to pass to the metric computations.
 
     Returns:
         A concatenated dataframe of the results of the case operator.
+
+    Raises:
+        TypeError: If any metric is not properly instantiated.
     """
     logger.info(
         "Computing case operator for case %s...",
         case_operator.case_metadata.case_id_number,
     )
+
+    # Validate that all metrics are instantiated (not classes or callables)
+    for i, metric in enumerate(case_operator.metric_list):
+        if isinstance(metric, type):
+            metric = metric()
+            case_operator.metric_list[i] = metric
```

Comment:
Not sure if this will be superseded by #228 but I had proposed the following:


```suggestion
            case_operator.metric_list[i] = metric()
```